Envisage and Apptools cron jobs failing to build traits

In both cases we are getting the following error:
```
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -O2 -fPIC -I/home/vagrant/workspace/Buildsystem/python/runtime/core/include/python3.8 -c traits/ctraits.c -o build/temp.linux-x86_64-cpython-38/traits/ctraits.o
  traits/ctraits.c:15:10: fatal error: Python.h: No such file or directory
     15 | #include "Python.h"
        |          ^~~~~~~~~~
  compilation terminated.
  error: command '/usr/bin/gcc' failed with exit code 1
  ----------------------------------------
  ERROR: Failed building wheel for traits
ERROR: Could not build wheels for traits, which is required to install pyproject.toml-based projects
Command '['/home/runner/edm/bin/edm', 'run', '-e', 'envisage-test-3.8-null', '--', 'python', '-m', 'pip', 'install', '--force-reinstall', 'git+http://github.com/enthought/traits.git#egg=traits']' returned non-zero exit status 1.
```

Failures [here](https://github.com/enthought/apptools/actions/runs/2930500430) and [here](https://github.com/enthought/envisage/actions/runs/2930457586).

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ['setuptools', 'wheel']
+requires = ['setuptools<65.2', 'wheel']
 build-backend = 'setuptools.build_meta'
 
 [tool.black]

